### DISCUSSION TITLE
Fix Flask run for applications without app.py

### DISCUSSION THREAD

**Discussion by thisisazeez** [pallets/flask#4339] [Q&A]
The issue is that when you want to run an app that doesn't have an app.py  file it will throw an error saying
Error: Could not locate a Flask application. You did not provide the "FLASK_APP" environment variable, and a "wsgi.py" # or "app.py" module was not found in the current directory.

This is my folder structure
![Screenshot from 2021-11-15 08-43-31](https://user-images.githubusercontent.com/86885841/141742171-e6bc0e0e-d879-4288-98e8-2f2275afa184.png)

The app is supposed to run even without app.py but if you do python controllers.py it will run on the port.
**The Error**
![Screenshot from 2021-11-15 08-43-08](https://user-images.githubusercontent.com/86885841/141742392-8ef05a37-1118-4bfd-bed3-6af79c8b61db.png)
Environment:

- Python version:3.10.0
- Flask version:3.8.10

**Comment by foarsitter** [comment]
Flask needs an app factory to start your application. By convention the app factory is located in `app.py`. Since you are not following this convention you need to update the `FLASK_APP` environment variable as show in the error message. 

Read more over here: https://flask.palletsprojects.com/en/2.0.x/patterns/appfactories/ 

I would like to ask you to close the issue since this question is not related to bugs in Flask itself. You will find better luck on getting help with your code on stackoverflow I think.

**Comment by ThiefMaster** [comment]
`FLASK_APP=controllers.py flask run` if that's where your `app` is defined (also make sure to NOT always acll `app.run()` in there)..
